### PR TITLE
[GOSDK-12971] Add example skeleton

### PR DESCRIPTION
**Pull request by iuliaromaniuc-tomtom**
[GOSDK-12971](https://jira.tomtomgroup.com/browse/GOSDK-12971)

### CODE REVIEW

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    private val progressUpdatedListener = ProgressUpdatedListener { routeProgress ->
+        val distanceAlongRoute = routeProgress.distanceAlongRoute
+        val remainingTimeOnRoute = routeProgress.remainingTime
+        Log.v("NavigationActivity", "distanceAlongRoute: " + distanceAlongRoute)
+        Log.v("NavigationActivity", "remainingTimeOnRoute: " + remainingTimeOnRoute)
+    }
+
+    companion object {
+        val amsterdam = GeoPoint(
```
Comment (AlinaUshakova-TomTom) [inline app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:115]: I think this should be one-liner at the place where you use it

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            latitude = 52.37616,
+            longitude = 4.90828
+        )
+
+        val rotterdam = GeoPoint(
```
Comment (AlinaUshakova-TomTom) [inline app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:120]: same here

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    private fun startNavigation() {
+        tomTomNavigation.addProgressUpdatedListener(progressUpdatedListener)
+
+        val routePlan = RoutePlan(route!!, routePlanningOptions!!)
+        val navigationOptions = NavigationOptions(routePlan)
+        tomTomNavigation.start(navigationOptions)        
+    }
+
+    private val progressUpdatedListener = ProgressUpdatedListener { routeProgress ->
+        val distanceAlongRoute = routeProgress.distanceAlongRoute
```
Comment (AlinaUshakova-TomTom) [inline app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:108]: maybe remove local vars, log directly `routeProgress.distanceAlongRoute`

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        initRouting()
+        planRoute(amsterdam, rotterdam)
+    }
+
+    private fun initRouting() {
+        routePlanner = OnlineRoutePlanner.create(context = this, apiKey = apiKey)
+        routeReplanner = OnlineRouteReplannerFactory.create(routePlanner)
```
Comment (AlinaUshakova-TomTom) [inline app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:50]: do we need `routeReplanner`?

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    private val apiKey = BuildConfig.TOMTOM_API_KEY
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        initRouting()
+        planRoute(amsterdam, rotterdam)
+    }
+
+    private fun initRouting() {
+        routePlanner = OnlineRoutePlanner.create(context = this, apiKey = apiKey)
```
Comment (AlinaUshakova-TomTom) [inline app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:49]: I think this function can go into `onCreate` if it left to one liner

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        routeReplanner = OnlineRouteReplannerFactory.create(routePlanner)
+    }
+
+    private fun planRoute(origin: GeoPoint, destination: GeoPoint) {
+        val itinerary = Itinerary(origin = origin, destination = destination)
+        val routePlanningOptions = RoutePlanningOptions(
+            itinerary = itinerary,
+            guidanceOptions = GuidanceOptions(
```
[the remark this answers is not in the document]
Comment (AlinaUshakova-TomTom) [reply app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:57]: Actually we can hide all of it. `GuidanceOptions` has default parameters.

`data class GuidanceOptions(
    val guidanceVersion: OnlineApiVersion = OnlineApiVersion.v1,
    val instructionType: InstructionType = InstructionType.Text,
    val language: Locale = Locale.ENGLISH,
    val roadShieldReferences: RoadShieldReferences = RoadShieldReferences.None,
    val announcementPoints: AnnouncementPoints = AnnouncementPoints.None,
    val phoneticsType: InstructionPhoneticsType = InstructionPhoneticsType.None,
    val extendedSections: ExtendedSections = ExtendedSections.None,
    val progressPoints: ProgressPoints = ProgressPoints.None
)`

[the remark this answers is not in the document]
Comment (AlinaUshakova-TomTom) [reply app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:57]: maybe `progressPoints` is something relevant, not sure

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+            context = this,
+            apiKey = apiKey,
+            locationProvider = locationProvider,
+            routeReplanner = routeReplanner
+        )
+        tomTomNavigation = TomTomNavigationFactory.create(navigationConfiguration)
+    }
+
+    private fun startNavigation() {
+        tomTomNavigation.addProgressUpdatedListener(progressUpdatedListener)
```
Comment (AlinaUshakova-TomTom) [inline app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:100]: move it to `initNavigation` ?

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt (hunk 1 above)
Comment (iuliaromaniuc-tomtom) [reply app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:115]: Yes, in the code snippets I will definitely put it right before where it is used.

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (iuliaromaniuc-tomtom) [reply app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:57]: Will check if the defaults are the same as what we use here.

File: app/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -16,6 +16,15 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.NavigatingWithTomTom"
         tools:targetApi="31">
+        <activity
```
[the remark this answers is not in the document]
Comment (iuliaromaniuc-tomtom) [reply app/src/main/AndroidManifest.xml:19]: It looks like we won't merge the PR at all.

File: app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt (hunk 4 above)
Comment (iuliaromaniuc-tomtom) [reply app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt:50]: But in retrospect, maybe we should just pass the route planner here. What is a route replanner, by the way?

**Pull request closed without merging**